calculator_test.go

```diff
@@ -2,23 +2,293 @@ package calculator_test
+			}
+		}
+		count++
+	}
+}
+
+func TestAddMultSub(t *testing.T) {
+	t.Parallel()
+	tcs := []struct {
+		operation func(float64, float64, ...float64) float64
+		name      string
+		arg1      float64
+		arg2      float64
+		nums      []float64
+		want      float64
+	}{
+		// add test cases
+		{
+			operation: calculator.Add,
+			name:      "Add: multiple numbers that sum to a positive",
+			arg1:      5,
+			arg2:      3,
+			nums:      []float64{2, 3, 4},
+			want:      17,
+		},
+		{
+			operation: calculator.Add,
+			name:      "Add: multiple numbers that sum to a negative",
+			arg1:      5,
+			arg2:      -3,
+			nums:      []float64{-2, -3, -4},
+			want:      -7,
+		},
+		// multiply test cases
+		{
+			operation: calculator.Multiply,
+			name:      "Negative number multiplied with a positive results in negative",
+			arg1:      -1,
+			arg2:      -1,
+			nums:      []float64{-1, 5},
+			want:      -5,
+		},
+		{
+			operation: calculator.Multiply,
+			name:      "Negative numbers multiplied result in a postive",
+			arg1:      1,
+			arg2:      1,
+			nums:      []float64{-3, -9},
+			want:      27,
+		},
+		{
+			operation: calculator.Multiply,
+			name:      "Number multiplied by zero results in zero",
+			arg1:      0,
+			arg2:      5,
+			nums:      []float64{5, 0},
+			want:      0,
+		},
+		// subtract test cases
+		{
+			operation: calculator.Subtract,
+			name:      "Numbers that subtract to a negative",
+			arg1:      -5,
+			arg2:      -5,
+			nums:      []float64{-1, 2},
+			want:      -1,
+		},
+		{
+			operation: calculator.Subtract,
+			name:      "Numbers that subtract to a positive",
+			arg1:      1000,
+			arg2:      800,
+			nums:      []float64{100, 99},
+			want:      1,
+		},
+		{
+			operation: calculator.Subtract,
+			name:      "Numbers that subtract to zero",
+			arg1:      9,
+			arg2:      3,
+			nums:      []float64{3, 3},
+			want:      0,
+		},
+	}
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			got := tc.operation(tc.arg1, tc.arg2, tc.nums...)
+			if !closeEnough(tc.want, got, 0.0001) {
+				t.Errorf("with args %f,%f,%v want %f, got %f", tc.arg1, tc.arg2, tc.nums, tc.want, got)
+			}
+		})
+	}
+}
+
+func TestDivide(t *testing.T) {
+	t.Parallel()
+	var testCases = []struct {
+		name        string
+		arg1        float64
+		arg2        float64
+		nums        []float64
+		want        float64
+		errExpected bool
+	}{
+		{
+			name:        "Number divided by zero results in an error",
+			arg1:        5,
+			arg2:        7,
+			nums:        []float64{6, 0},
+			want:        0,
+			errExpected: true,
+		},
+		{
+			name: "Number divided by another number results in an integer result",
+			arg1: 12,
+			arg2: 2,
+			nums: []float64{2, 3},
+			want: 1,
+		},
+		{
+			name: "Number divided by another number results in a rational result",
+			arg1: 1000,
+			arg2: 10,
+			nums: []float64{10, 4},
+			want: 2.5,
+		},
+		{
+			name: "Incorrect division but no err expected",
```

Comment:
Why is it 'incorrect'?

Reply:
Ah, I left this in on accident.  I was testing the errExpected case.  Meant to delete it but I just changed the value back to where it worked instead.  Removed.